### PR TITLE
[4.0] Implementation of feature 'filter menu items by component'

### PR DESCRIPTION
**Pull request by Tchangue**
Pull Request for this Issue [here](https://github.com/joomla/joomla-cms/pull/21358 "")

### Summary of Changes
We built a filter for menu items by components so that we can have now the possibility to filter items or subitems that belongs to a special component.

### Testing Instructions
After creating new menu items (of same and different component types) in differents menus you click on the dropdown menu "Table options" and select the menu " Select Component Menu Item " and then you can choose which items of which component you want to display.
You should test on the site backend (not the administrator backend)

### Expected result
Items of the components choosed must be showed.

### Actual result
Items of the components choosed are showed.

### CODE REVIEW

File: administrator/language/en-GB/en-GB.com_menus.ini [hunk 1]
```diff
@@ -193,3 +193,7 @@ COM_MENUS_VIEW_NEW_ITEM_TITLE="Menus: New Item"
 COM_MENUS_VIEW_NEW_MENU_TITLE="Menus: Add"
 COM_MENUS_XML_DESCRIPTION="Component for creating menus."
 JLIB_RULES_SETTING_NOTES="Changes apply to this component only.<br><em><strong>Inherited</strong></em> - a Global Configuration setting or higher level setting is applied.<br><em><strong>Denied</strong></em> always wins - whatever is set at the Global or higher level and applies to all child elements.<br><em><strong>Allowed</strong></em> will enable the action for this component unless overruled by a Global Configuration setting."
+COM_MENUS_FILTER_COMPONENT_MENU_ITEM_DESC="Filters the menu items list where the component is this selected menu item."
+COM_MENUS_FILTER_COMPONENT_MENU_ITEM_LABEL="Component Menu Item"
+COM_MENUS_FILTER_COMPONENT_MENU_ITEM_DESC="Filters the menu items list where the component is this selected menu item."
```
Comment (brianteeman) [inline administrator/language/en-GB/en-GB.com_menus.ini]: duplicate

**Comment by ghost** [comment]
Changed Title to show its about 4.0.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21358">issues.joomla.org/tracker/joomla-cms/21358</a>.</sub>

**Comment by ghost** [comment]
I have tested this item :red_circle: unsuccessfully on add10b5ff7934d9e6e58de5a7143dee233d125fe<br /><br />After click on "Table Options" got:

`/administrator/components/com_menus/Field/MenuItemByComponentField.php on line 284`<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21358">issues.joomla.org/tracker/joomla-cms/21358</a>.</sub>

**Comment by wojsmol** [comment]
@Tchangue Please see codestyle issues http://ci.joomla.org/joomla/joomla-cms/8026/8

**Comment by SharkyKZ** [comment]
Need to account for system links. Their value is shown instead of title because they do not have `componentname`.

**Comment by ghost** [comment]
Closed as abandoned.

**Pull request closed without merging**